src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,117 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int SIZE_MULTIPLICATOR = 2;
+
+    private int currentCapacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        table = new Node[currentCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (keyExists(key)) {
+            updateValue(key, value);
+            return;
+        }
+        if (size + 1 > currentCapacity * DEFAULT_LOAD_FACTOR) {
```

Comment:
I think it is better to extract 'curentCapacity * DEFAULT_LOAD_FACTOR ' in a variable (for example - threshold).

Reply:
done